src/Microsoft.SqlTools.ServiceLayer/ObjectExplorer/SmoModel/SmoQueryModel.cs

```diff
@@ -19,15 +21,33 @@ internal partial class SqlDatabaseQuerier: SmoQuerier
 
         public override Type[] SupportedObjectTypes { get { return supportedTypes; } }
 
-        public override  IEnumerable<SqlSmoObject> Query(SmoQueryContext context)
+        public override  IEnumerable<SqlSmoObject> Query(SmoQueryContext context, string filter)
         {
             Server parentServer = context.Parent as Server;
             if (parentServer != null)
             {
                 var retValue = parentServer.Databases;
-                if(retValue != null)
+                bool hasFilter = !string.IsNullOrEmpty(filter);
+                HashSet<string> urns = null;
+                if (hasFilter)
                 {
-                    return new SmoCollectionWrapper<Database>(retValue);
+                    string urn = $"{parentServer.Urn.ToString()}/Database" + filter;

```

Comment:
This is a great change. Nice job using SFC to get efficient querying while still returning usable SMO objects so we can cleanly manage the code later on.